webxrlayers-1.bs

```diff
@@ -0,0 +1,730 @@
+
+ISSUE: how to do hit testing? Should it be done by the UA?
+
+</section>
+
+
+Terminology {#terminology}
+-----------
+
+Application flow {#applicationflow}
+----------------
+
+<section class="non-normative">
+
+If an author wants to use layers, they have to go through these steps:
+ 1. request support through {{XRPermissionDescriptor/requiredFeatures}} or {{XRPermissionDescriptor/optionalFeatures}} in {{XRSystem/requestSession()}}
+ 1. create a XRWebGLLayerFactory or XRMediaLayerFactory
+ 1. create layers with these factories
+ 1. add the layers to {{XRRenderStateInit}} and call {{XRSession/updateRenderState()}}
+ 1. during {{XRSession/requestAnimationFrame()}} for webgl layers, draw content each gl layer
+
+</section>
+
+Model {#model}
+======
+
+In order for the applications to signal their interest in using layers during a session, 
+the session must be requested with appropriate [=feature descriptor=]. The string <dfn>layers</dfn> is introduced 
+by this module as a new valid feature descriptor for WebXR Layers feature.
+
+<div class="example">
+The following code requests layers as an optional feature.
+
+<pre highlight="js">
+navigator.xr.requestSession('immersive-vr', {
+    optionalFeatures: ['layers']
+  }
+</pre>
+</div>
+
+Layers are only supported for XRSessions created with XRSessionMode of {{XRSessionMode/"immersive-vr"}} 
+or {{XRSessionMode/"immersive-ar"}}. {{XRSessionMode/"inline"}} sessions must not support layers.
+
+Layer types {#xrlayertypes}
+===========
+
+Mono and stereo layers {#monovsstereo}
+----------------------
+A stereo layer must supply a different view to render to for each eye. Stereo layers should be refreshed 
+close to the device's native frame rate.
+
+A mono layers must supply a single view which is shown to each eye. There is no requirement to refresh mono 
+layers repeatedly.
+
+The [=XR Compositor=] must ensure that layers are presented correctly in stereo to the observer.
+
+XRLayer {#xrlayertype}
+-------
+Every layer except XRWebGLLayer is derived from XRLayer.
+XRLayer defines a set of common attributes and behaviors acress all layer types.
+
+<pre class="idl">
+interface XRLayer {
+  readonly attribute unsigned long pixelWidth;
+  readonly attribute unsigned long pixelHeight;
+
+  attribute boolean blendTextureSourceAlpha;
+  attribute boolean chromaticAberrationCorrection;
+
+  void destroy();
+};
+</pre>
+
+The <dfn attribute for="XRLayer">pixelWidth</dfn> and <dfn attribute for="XRLayer">pixelHeight</dfn> attributes 
+return the width and height of the GL attachments, respectively. If there are no attachments, they will return 0.
+
+The <dfn attribute for="XRLayer">blendTextureSourceAlpha</dfn> attribute enables the layer’s texture alpha channel.
+
+The <dfn attribute for="XRLayer">chromaticAberrationCorrection</dfn> attribute enables optical chromatic 
```

Comment:
We should somehow indicate that this is more of a hint and that the compositor may not be able to respect it, since various systems may either force it on or not support it at all.

Reply:
done. Also I will add an issue as this attribute needs a better description.